src/hooks/Tokens.ts

```diff
@@ -10,31 +10,33 @@ import { isL2ChainId } from 'utils/chains'
 
 import { useAllLists, useCombinedActiveList, useCombinedTokenMapFromUrls } from '../state/lists/hooks'
 import { WrappedTokenInfo } from '../state/lists/wrappedTokenInfo'
-import { useUserAddedTokens, useUserAddedTokensOnChain } from '../state/user/hooks'
+import { useUserAddedTokens } from '../state/user/hooks'
 import { TokenAddressMap, useUnsupportedTokenList } from './../state/lists/hooks'
 
 // reduce token map into standard address <-> Token mapping, optionally include user added tokens
-function useTokensFromMap(tokenMap: TokenAddressMap): { [address: string]: Token } {
-  const { chainId } = useWeb3React()
+function useTokensFromMap(tokenMap: TokenAddressMap, chainId?: SupportedChainId): { [address: string]: Token } {
+  const { chainId: connectedChainId } = useWeb3React()
   return useMemo(() => {
-    if (!chainId) return {}
+    if (!chainId && !connectedChainId) return {}
```

Comment:
why `chainId` is optional argument to `useTokensFromMap` if we seem to return from the function ALWAYS when it's not defined? We should mark it as required? We already have some issues with hooks-based APIs, but I think we're all used to using ugly ternary operators at the call-site, instead of accepting undefined as arguments and handling that inside a hook.

Reply:
@grabbou we return early when both `chainId` *and* `useWeb3React().chainId` are undefined. the argument is optional because we fallback to the connected chain Id in this function.

do you think i should remove the fallback, and move the `useWeb3React()` out of this function to each callsite?

Reply:
ah nevermind, i see your other comment now!